### PR TITLE
Core/Scripts: Fix Blood pool summon and position

### LINKED ISSUE #13162: [3.3.5][6.x][Core|Spell] Animal Blood
core 961aaf5
db up to date

If you have the aura http://www.wowhead.com/spell=46221 and if you walk into water, you will cast this spell: SPELL_SPAWN_BLOOD_POOL = 63471 in order to create the Blood-GO.

Problem:
The GO: http://de.wowhead.com/object=194479 should always be direktly on the water. Right now the go is summoned sometimes over the water and sometimes under the water - if i walked into the water.

http://www.pic-upload.de/view-24643579/WoWScrnShot_091814_112738.jpg.html
https://github.com/TrinityCore/TrinityCore/blob/7d1d55124c1e289748d2afefae896c97f710e57b/src/server/scripts/Spells/spell_generic.cpp#L195

prio: low
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/4460494-core-spell-animal-blood?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sirikfoll**
**Changes proposed:**

-  Use GetLiquidStatus to determine correct Z position for GameObject summon, should be at water level.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Issues addressed:** Closes #13162

**Tests performed:** Build and tested (.cast 46221, and enter inside water)

### CODE REVIEW

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -189,9 +190,11 @@ class spell_gen_animal_blood : public AuraScript
 
     void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
     {
+        if (GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_EXPIRE)
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_generic.cpp]: expire isnt the only reason for this aura to be removed outside of water (interrupt flag)
immunity (divine shield?)
death

Comment (sirikfoll) [reply src/server/scripts/Spells/spell_generic.cpp]: Although the spell IsDeathPersistent, it seems to be broken and gets removed, but code triggers on logout anyway

Comment (sirikfoll) [reply src/server/scripts/Spells/spell_generic.cpp]: Not sure what to do here, to match the exact desired behavior, the calls to `RemoveAurasWithInterruptFlags(AURA_INTERRUPT_FLAG_NOT_ABOVEWATER)` needs to be changed.

From `Unit::ProcessTerrainStatusUpdate` (triggers by stepping on water, shallow or not)
To `Player::SetInWater(bool apply)` (triggers only when "swimming") <- This is what seems blizzlike

**Comment by illfated** [comment]
**Conflicting files**
`sql/updates/world/3.3.5/2019_03_04_03_world.sql`

... the disadvantage of using SQL file names which could be used before the PR gets merged.

File: src/server/scripts/Spells/spell_generic.cpp (hunk 1 above)
Comment (jackpoz) [reply src/server/scripts/Spells/spell_generic.cpp]: how is the status of this ?

Comment (sirikfoll) [reply src/server/scripts/Spells/spell_generic.cpp]: Decided to just merge the missing script, the remaining issue is due to generic changes that I'm not confident enough to touch, like #23091 and https://github.com/TrinityCore/TrinityCore/pull/23088#discussion_r262303442

**Pull request merged**